my_sprites.py

```diff
@@ -417,163 +517,35 @@ def __init__(
-
-    @property
-    def hp(self):
-        return self._hp
+        self.health_bar = HealthBar(max_health=self.max_hp)
 
     @property
     def is_walking(self):
         return True in [self.left_pressed, self.up_pressed, self.right_pressed, self.down_pressed]
 
-    @hp.setter
-    def hp(self, new_hp):
-        self._hp = max(0, min(new_hp, self.max_hp))  # hp should be greater than 0 and not greater than max hp
-
-
-    @property
-    def weapons(self):
-        """
-        A list of weapon types carried by the player
-        """
-        return self._weapons.keys()
-
-    @property
-    def max_hp(self):
-        return self._max_hp
-
-    @property
-    def emotes(self):
-        return self._emotes
-
-    @property
-    def type(self):
-        return self._type
-
-    @property
-    def equiped(self):
-        return self._equiped
-
-    @equiped.setter
-    def equiped(self, weapon):
-        assert type(weapon) == Weapon or weapon is None, f"expected type Weapon or NoneType, got {type(weapon)}"
-        self._equiped = weapon
-
-    @property
-    def direction(self):
-        return self._direction
-
-    def add_weapon(self, type):
-        """
-        Add a weapon to the player's weapons.
-        """
-
-        # If we already have the weapon type,
-        # it will be replaced by the new weapon.
-        self._weapons[type] = Weapon(type)
-
-        # If no weapon is in use, start using the one that was picked up
-        if self.equiped is None:
-            self.equip(type)
-
-    def equip(self, type):
-        """
-        The player will start using this weapon type if available.
-        This should probably be all type of objects
-        Return value reports success.
-        """
-        if type in self._weapons.keys():
-            self._equiped = self._weapons[type]
-            return True
-
-        # Could not equip the weapon type
-        return False
-
     def on_key_press(self, key, modifiers):
         """
         Track the state of the control keys
         """
 
-        previous_direction = self._direction
-
         if key == self.key_left:
             self.left_pressed = True
             # Turns the sprite to the left side.
             self.texture = self.textures[1]
-            self._direction = Direction.LEFT
+            self.facing_dir = Direction.LEFT
             return
         elif key == self.key_right:
             self.right_pressed = True
             # Turns the sprite to the Right side
             self.texture = self.textures[0]
-            self._direction = Direction.RIGHT
+            self.facing_dir = Direction.RIGHT
             return
         elif key == self.key_up:
             self.up_pressed = True
-            self._direction = Direction.UP
+            self.facing_dir = Direction.UP
         elif key == self.key_down:
             self.down_pressed = True
-            self._direction = Direction.DOWN
+            self.facing_dir = Direction.DOWN
```

Comment:
Why do you want to rename them and make them non _"private"_ (removing _ prefix) ?

Reply:
so, just rename the variable to _facing_dir?
- not make it into a proberty